scene/resources/motion_blur_compositor_effect.cpp

```diff
@@ -0,0 +1,290 @@
+	push_constant[2] = motion_blur_center_fade;
+	push_constant[3] = 0.0f;
+
+	int view_count = render_scene_buffers->get_view_count();
+	for (int view = 0; view < view_count; view++) {
+		RID color_image = render_scene_buffers->call("get_color_layer", view);
+		RID velocity_image = render_scene_buffers->call("get_velocity_layer", view);
+		RID texture_image = render_scene_buffers->get_texture_slice(context, texture, view, 0, 1, 1);
+		RenderingDevice::get_singleton()->draw_command_begin_label("Compute blur " + itos(view), Color(1.0, 1.0, 1.0, 1.0));
+		Vector<RenderingDevice::Uniform> uniforms;
+		uniforms.push_back(get_sampler_uniform(color_image, 0));
+		uniforms.push_back(get_sampler_uniform(velocity_image, 1));
+		uniforms.push_back(get_image_uniform(texture_image, 2));
+
+		RID tex_uniform_set = RD::get_singleton()->uniform_set_create(uniforms, motion_blur_shader, 0);
+
+		int x_groups = ((render_size.x - 1) / 8) + 1;
+		int y_groups = ((render_size.y - 1) / 8) + 1;
+
+		RenderingDevice::ComputeListID compute_list = RD::get_singleton()->compute_list_begin();
+		RD::get_singleton()->compute_list_bind_compute_pipeline(compute_list, motion_blur_pipeline);
+		RD::get_singleton()->compute_list_bind_uniform_set(compute_list, tex_uniform_set, 0);
+		RD::get_singleton()->compute_list_set_push_constant(compute_list, push_constant.ptr(), push_constant.size() * sizeof(float));
+		RD::get_singleton()->compute_list_dispatch(compute_list, x_groups, y_groups, 1);
+		RD::get_singleton()->compute_list_end();
+
+		RD::get_singleton()->draw_command_end_label();
+
+		RD::get_singleton()->draw_command_begin_label("Overlay result " + itos(view));
+
+		uniforms.clear();
+		uniforms.push_back(get_sampler_uniform(texture_image, 0));
+		uniforms.push_back(get_image_uniform(color_image, 1));
+
+		tex_uniform_set = RD::get_singleton()->uniform_set_create(uniforms, overlay_shader, 0);
+
+		compute_list = RD::get_singleton()->compute_list_begin();
+		RD::get_singleton()->compute_list_bind_compute_pipeline(compute_list, overlay_pipeline);
+		RD::get_singleton()->compute_list_bind_uniform_set(compute_list, tex_uniform_set, 0);
+		RD::get_singleton()->compute_list_dispatch(compute_list, x_groups, y_groups, 1);
+		RD::get_singleton()->compute_list_end();
+
+		RD::get_singleton()->draw_command_end_label();
+	}
+
+	RD::get_singleton()->draw_command_end_label();
+}
+
+MotionBlurCompositorEffect::~MotionBlurCompositorEffect() {
+	if (linear_sampler.is_valid()) {
+		RD::get_singleton()->free(linear_sampler);
+	}
+	if (motion_blur_shader.is_valid()) {
+		RD::get_singleton()->free(motion_blur_shader);
+	}
+	if (overlay_shader.is_valid()) {
+		RD::get_singleton()->free(overlay_shader);
+	}
+}
+
+MotionBlurCompositorEffect::MotionBlurCompositorEffect() {
+	set_effect_callback_type(EFFECT_CALLBACK_TYPE_POST_TRANSPARENT);
+	set_needs_motion_vectors(true);
+	if (!DisplayServer::get_singleton()->window_can_draw()) {
+		return;
+	}
+	_initialize_compute();
```

Comment:
```suggestion
	RenderingServer::get_singleton()->call_on_render_thread(callable_mp(this, &MotionBlurCompositorEffect::_initialize_compute));
```

Was previously this.